cmake/Modules/FindEigen3.cmake

```diff
@@ -62,25 +65,43 @@ if (EIGEN3_INCLUDE_DIR)
   _eigen3_check_version()
   set(EIGEN3_FOUND ${EIGEN3_VERSION_OK})
 
-else (EIGEN3_INCLUDE_DIR)
+endif (EIGEN3_INCLUDE_DIR)
+
+if(NOT EIGEN3_INCLUDE_DIR)
+  # search first if an Eigen3Config.cmake is available in the system,
+  # if successful this would set EIGEN3_INCLUDE_DIR and the rest of
+  # the script will work as usual
+  find_package(Eigen3 ${Eigen3_FIND_VERSION} NO_MODULE QUIET
+      PATHS
+        /usr/lib/${CMAKE_LIBRARY_ARCHITECTURE}/cmake
+  )
+endif(NOT EIGEN3_INCLUDE_DIR)
 
+if(NOT EIGEN3_INCLUDE_DIR)
   find_path(EIGEN3_INCLUDE_DIR NAMES signature_of_eigen3_matrix_library
       HINTS
-      ${EIGEN3_ROOT}
+      ENV EIGEN3_ROOT
```

Comment:
This is redundant with the CMake policy [CMP0074](https://cmake.org/cmake/help/v3.12/policy/CMP0074.html), which is OLD by default so any `<PackageName>_ROOT` (env or CMake) is ignored before checking the HINTS and prints **a lot** of warnings.
I would agree setting this policy to NEW in SOFA. It would also make this hint unnecessary.